[Updated Eval] math_logic_operations

nathanstew7 opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑

### Eval name

math_logic_operations

### Eval description

This eval redefines several basic mathematical rules concerning the standard "order of operations," or BEDMAS. The LLM is instructed to address addition and subtraction operations before multiplication and division, and it is also advised to interpret any operations within parentheses () as the "opposite" operation - that is, addition in parentheses is treated like subtraction, multiplication in parentheses is treated like division, and so on. The LLM is then asked a mathematical question, such as 4*5-3, and asked to give a numerical answer. This answer is evaluated based on whether it adheres to the new rules laid out.

### What makes this a useful eval?

It is well known that LLMs currently struggle with even basic logic, reasoning, and mathematics. GPT-4, while it does better than many other LLMs, is still not largely an exception. Many groups are working to enhance LLMs' logical problem-solving capabilities, and this eval combines both mathematical and logical reasoning. Logically, the LLM must be able to interpret the new rules laid out for it, and since they are mathematical rules, it must be able to apply them to its existing mathematical framework.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

This eval includes 30 examples. Please let me know if I can change or add anything!

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 5+7","name":"example_user"},{"role":"system","content":"A: 12","name":"example_assistant"},{"role":"user","content":"Q: 5+7"}],"ideal":["A: 12"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 5+7*4","name":"example_user"},{"role":"system","content":"A: 48","name":"example_assistant"},{"role":"user","content":"Q: 5+7*4"}],"ideal":["A: 48"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 6*4-3","name":"example_user"},{"role":"system","content":"A: 6","name":"example_assistant"},{"role":"user","content":"Q: 6*4-3"}],"ideal":["A: 6"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 8/4-2","name":"example_user"},{"role":"system","content":"A: 4","name":"example_assistant"},{"role":"user","content":"Q: 8/4-2"}],"ideal":["A: 4"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 0*1+8","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 0*1+8"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (9+9)","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: (9+9)"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1+(1-1)","name":"example_user"},{"role":"system","content":"A: 3","name":"example_assistant"},{"role":"user","content":"Q: 1+(1-1)"}],"ideal":["A: 3"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 4+4*4-4","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 4+4*4-4"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 7*9-1/7","name":"example_user"},{"role":"system","content":"A: 8","name":"example_assistant"},{"role":"user","content":"Q: 7*9-1/7"}],"ideal":["A: 8"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 9*9-9*9","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 9*9-9*9"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 8*8-8/8","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 8*8-8/8"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1+1-1*1+1-1","name":"example_user"},{"role":"system","content":"A: 1","name":"example_assistant"},{"role":"user","content":"Q: 1+1-1*1+1-1"}],"ideal":["A: 1"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 0*0*0+1","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 0*0*0+1"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1-2*2","name":"example_user"},{"role":"system","content":"A: -2","name":"example_assistant"},{"role":"user","content":"Q: 1-2*2"}],"ideal":["A: -2"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 4*3-5","name":"example_user"},{"role":"system","content":"A: -8","name":"example_assistant"},{"role":"user","content":"Q: 4*3-5"}],"ideal":["A: -8"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1+1+1+1*5","name":"example_user"},{"role":"system","content":"A: 20","name":"example_assistant"},{"role":"user","content":"Q: 1+1+1+1*5"}],"ideal":["A: 20"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 2*1-5/4","name":"example_user"},{"role":"system","content":"A: -2","name":"example_assistant"},{"role":"user","content":"Q: 2*1-5/4"}],"ideal":["A: -2"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (5-100)/110-5","name":"example_user"},{"role":"system","content":"A: 1","name":"example_assistant"},{"role":"user","content":"Q: (5-100)/110-5"}],"ideal":["A: 1"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 32-28/(2-1-1)-(0-0)","name":"example_user"},{"role":"system","content":"A: 1","name":"example_assistant"},{"role":"user","content":"Q: 32-28/(2-1-1)-(0-0)"}],"ideal":["A: 1"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (4-(1-4))","name":"example_user"},{"role":"system","content":"A: 9","name":"example_assistant"},{"role":"user","content":"Q: (4-(1-4))"}],"ideal":["A: 9"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 9-(9)","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: 9-(9)"}],"ideal":["A: 0"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 9-(9-9)","name":"example_user"},{"role":"system","content":"A: -9","name":"example_assistant"},{"role":"user","content":"Q: 9-(9-9)"}],"ideal":["A: -9"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1+2-3+4*5/(9+8)","name":"example_user"},{"role":"system","content":"A: 20","name":"example_assistant"},{"role":"user","content":"Q: 1+2-3+4*5/(9+8)"}],"ideal":["A: 20"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 1+(1+(1+(1)))","name":"example_user"},{"role":"system","content":"A: 2","name":"example_assistant"},{"role":"user","content":"Q: 1+(1+(1+(1)))"}],"ideal":["A: 2"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 7*9-9+2-(3-3)+1","name":"example_user"},{"role":"system","content":"A: -21","name":"example_assistant"},{"role":"user","content":"Q: 7*9-9+2-(3-3)+1"}],"ideal":["A: -21"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 2+2/2+9-(8+8)-(2/0)-7","name":"example_user"},{"role":"system","content":"A: 1","name":"example_assistant"},{"role":"user","content":"Q: 2+2/2+9-(8+8)-(2/0)-7"}],"ideal":["A: 1"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (9/8*72)+(100+99)/100-99+(2*2)","name":"example_user"},{"role":"system","content":"A: 1","name":"example_assistant"},{"role":"user","content":"Q: (9/8*72)+(100+99)/100-99+(2*2)"}],"ideal":["A: 1"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: 6-(5*5)/(5-5)-(90*10)","name":"example_user"},{"role":"system","content":"A: 5","name":"example_assistant"},{"role":"user","content":"Q: 6-(5*5)/(5-5)-(90*10)"}],"ideal":["A: 5"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (70-1)/(10-10-10-10-10-10-10-1)-70","name":"example_user"},{"role":"system","content":"A: 71","name":"example_assistant"},{"role":"user","content":"Q: (70-1)/(10-10-10-10-10-10-10-1)-70"}],"ideal":["A: 71"]}
{"input":[{"role":"system","content":"You currently do math using the standard order of operations. I am going to redefine this order of operations, and introduce you to some new rules. Then you will answer some math questions. From now on, execute addition (+) and subtraction (-) before multiplication (*) and division (/) (e.g. 5*4+2 would now be treated as 5*6=30, since you evaluate 4+2=6 first, and then multiply 5*6), and interpret any operations inside parentheses as the 'opposite' operation. Addition in parentheses should be interpreted as subtraction and multiplication in parentheses should be interpreted as division, and vice versa. Continue to compute operations inside parentheses before other operations. Using these rules, answer the following question with only a numerical response (do not explain your thinking, but make sure to think step-by-step and consider these instructions as you formulate your response). Answer with only a number, nothing else."},{"role":"system","content":"Q: (0/0)","name":"example_user"},{"role":"system","content":"A: 0","name":"example_assistant"},{"role":"user","content":"Q: (0/0)"}],"ideal":["A: 0"]}

  ```
</details>

nathanstew7 commented:
Hello, I received a comment that I should revert changes to the .gitattributes file, and only track the .JSONL file with Git LFS. This updated PR addresses those requests - my apologies! Thank you!

nathanstew7 commented:
Hi Tim, my apologies, but I cannot re-open the previous PR. Its head repository was deleted when I forked the OpenAI evals repo to create this updated PR.

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request the following changes:

1. I would recommend asking the model to reason before answering and using the `Includes` evaluation method here because logical reasoning and other complex tasks are hard for the model to do zero-shot without a chance to reason through the steps. Asking the model to provide reasoning will give the model a fair chance to solve the question. You can also add instructions to provide output in a specific format and then use that format to write the ideal answer. For example, the model can be asked to enclose the final answer in square brackets, and the ideal answer can be formatted like `[5]`. It'll give the model a proper chance to reason before answering, and the proper formatting will help in identifying the final answer using the `Includes` method.

2. For all the samples, the sample question and answer are the same as the actual question, which means you are already providing the answer to the model. Kindly remove this.

We would love to review the PR again after the suggested changes.

nathanstew7 commented:
Hi @usama-openai, thank you for getting back to me! I have switched to the `Includes` evaluation method, asked the model to reason, and specified that I want the model to put its answer in square brackets, as you advised. 

I also removed the answers from the question itself - I hadn't realized this was being fed in as part of the prompt! 

This PR should be ready to review again.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

nathanstew7 commented:
I would really appreciate access to the 8k and 32k GPT-4 APIs, and am happy to do more evals if that helps!

usama-openai commented:
You should see GPT-4 API access enabled in your account in the next few days.

nathanstew7 commented:
Hello @usama-openai @andrew-openai, I would really appreciate it if someone would activate the GPT-4 API for my account. Thank you.

usama-openai commented:
Can you try https://platform.openai.com/playground?mode=chat&model=gpt-4?
